### PR TITLE
EVG-7160 Use LoadProjectInto instead of ReadYAMLInto in fetch

### PR DESCRIPTION
**Pull request by bsamek**
Reading a project YAML into a model.Project is not always valid. Tags cannot be
parsed correctly. Instead it has to be parsed with LoadProjectInto.

### CODE REVIEW

**Comment by bsamek** [comment]
I tested this by running `evergreen fetch` against the problematic YAML from MCI-3127, which fails with the current CLI and passes with these changes.

**Comment by bsamek** [comment]
Added @ablack12 in case fetch needs to change its behavior when we switch completely over to ParserProject, though I imagine if that's the case that should be in a separate ticket.

File: operations/http.go [hunk 1]
```diff
@@ -259,7 +259,11 @@ func (ac *legacyClient) GetConfig(versionId string) (*model.Project, error) {
 		return nil, NewAPIError(resp)
 	}
 	ref := &model.Project{}
-	if err := util.ReadYAMLInto(resp.Body, ref); err != nil {
```
Comment (ybrill) [inline operations/http.go]: does [this](https://github.com/evergreen-ci/evergreen/blob/943224c9b4d87ee64af38f8ae946004330ec9a16/operations/http.go#L246) need to be changed as well?

Comment (bsamek) [reply operations/http.go]: yup, fixed

**Pull request merged**